bin/node-gyp.js

```diff
@@ -16,11 +16,11 @@ var completed = false
 prog.parseArgv(process.argv)
 prog.devDir = prog.opts.devdir
 
-var homeDir = osenv.home()
+var homeDir = os.homedir()
 if (prog.devDir) {
   prog.devDir = prog.devDir.replace(/^~/, homeDir)
 } else if (homeDir) {
-  prog.devDir = path.resolve(homeDir, '.node-gyp')
+  prog.devDir = envPaths('node-gyp', { suffix: '' }).cache
```

Comment:
The suffix is 'nodejs' by default, another option is: `envPaths('gyp').cache`.
I opted for `.cache` because I think it fits the type of data more than `.data` since it's not essential data.